rosetta-source/src/main/rosetta/event-position-type.rosetta

```diff
@@ -42,3 +48,35 @@ type Portfolio: <" A Portfolio represents an aggregation of multiple Positions,
 
 	aggregationParameters AggregationParameters (1..1) <"Describes the portfolio by describing how to aggregate all its relevant Events.">
 	portfolioState PortfolioState (1..1) <"Describes the state of the Portfolio as a list of Positions resulting from the aggregation.">
+
+type InventoryBase: <"A data type that can be used to describe an inventory of securities.">
+    inventoryRecord InventoryRecord (0..*) <"An array holding the list of inventory being described. Each element in the inventoryRecord array represents an individual piece of inventory i.e. a security.">
+
+type InventoryRecordBase: <"An individual piece of inventory. This represents a single security.">
+    identifer AssignedIdentifier (1..1) <"Unique identifier for this record. This can be used to uniquely identify a specific piece of inventory.">
+    security Security (1..1) <"The security details.">
+
+type Inventory extends InventoryBase: <"Describes an inventory of securities.">
+
+type InventoryRecord extends InventoryRecordBase: <"Describes an individual piece of inventory.">
+
+type AvailableInventory: <"A data type that can be used to describe the inventory of securities that a party holds. ">
+    party Party (0..*) <"Defines all parties involved for the list of inventory records in this set of inventory. For example, when used to describe securities lending availability, this could hold the sender of the availability, the intended recipient, the beneficial owner(s), the lender (which may differ from the sender as the lender may have the same piece of availability going through multiple agents), an agent or a venue.">
+    partyRole PartyRole (0..*) <"Defines the role(s) that party(ies) may have in relation to the inventory.">
```

Comment:
I understand the description of `PartyRole` in `AvailableInventoryRecord`, that it needs to be defined at the security level. However in this case we shouldn't duplicate `PartyRole` in `AvailableInventory` as well. Or are we saying that there are certain party roles that apply at the `AvailableInventory` level, and some at the `AvailableInventoryRecord` level? In which case you may want to restrict the set of enum values that would be allowed at each level, or at least make sure that they are mutually exclusive.